### ISSUE TITLE
`ssr-only-secrets` has no exported member named `cloakSSROnlySecret`

### ISSUE THREAD

**Issue by AlSirang** [phryneas/ssr-only-secrets#1]
When trying to import `cloakSSROnlySecret` in a Next.js server component, I am getting the following error:

_`ssr-only-secrets has no exported member named cloakSSROnlySecret`_

**Here is the component:**

```js
import { Metadata } from 'next'
import { Providers } from '@/src/lib/providers'
import './globals.css'
import { ApolloWrapper } from '../lib/configs/apolloWraper'
import { cookies } from 'next/headers'
import { cloakSSROnlySecret } from 'ssr-only-secrets'
import authConfig from '../lib/configs/authConfig'

export default function RootLayout({ children }: { children: React.ReactNode }) {
  const sessionIdPlain = cookies().get(authConfig.storageTokenKeyName)?.value

  const sessionId = cloakSSROnlySecret(sessionIdPlain ?? '', 'SECRET_KEY')
  return (
    <html lang='en'>
      <body>
        <ApolloWrapper sessionId={sessionId}>
          <Providers>{children}</Providers>
        </ApolloWrapper>
      </body>
    </html>
  )
}

```
**Versions:**
```
 "next": "14.0.1",
 "ssr-only-secrets": "^0.1.1",
 "typescript": "^5"
 ```
I am using the Next.js app router, and the above code is from RootLayout. Do I need to configure something else to resolve this error? The cloakSSROnlySecret function works, but I'm getting an error when I import it.

**tsconfig.json**

```js
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "bundler",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@/*": ["./*"]
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
  "exclude": ["node_modules"]
}
```

**Comment by phryneas** [comment]
`cloakSSROnlySecret` is only available from React Server Components. It seems like your root layout is a RSC, but does maybe some other file that contains a Client Component import something from it?

**Comment by phryneas** [comment]
Try to add a `import "server-only"` at the top of your RootLayout to see if that triggers - that would validate that you're not in a RSC.

**Comment by AlSirang** [comment]
I added `import "server-only"` but it did nothing (no errors or warnings). I also removed all imports and only imported `cloakSSROnlySecret`, but it is still giving me an error for the import.

RootLayout with all imports removed:



```js


import 'server-only'

import { cookies } from 'next/headers'
import { cloakSSROnlySecret } from 'ssr-only-secrets'

import './globals.css'



export default function RootLayout({ children }: { children: React.ReactNode }) {
  const sessionIdPlain = cookies().get('access_token')?.value

  const sessionId = cloakSSROnlySecret(sessionIdPlain ?? '', 'SECRET_KEY')
  return (
    <html lang='en'>
      <body>
        {children}
      </body>
    </html>
  )
}
```

**Comment by AlSirang** [comment]
I just created a brand new Next.js app, but it still gives me the error mentioned above.

Versions:

- "next": "^14.2.3"
- "react": "^18.3.1"
- "react-dom": "^18.3.1"
- "ssr-only-secrets": "^0.1.1"

**RootLayout.tsx**

```js
import type { Metadata } from "next";
import { Inter } from "next/font/google";
import "./globals.css";

import { cookies } from "next/headers";

// '"ssr-only-secrets"' has no exported member named 'cloakSSROnlySecret'. Did you mean 'readSSROnlySecret'?ts(2724)
import { cloakSSROnlySecret } from "ssr-only-secrets";

const inter = Inter({ subsets: ["latin"] });

export const metadata: Metadata = {
  title: "Create Next App",
  description: "Generated by create next app",
};

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  const sessionIdPlain = cookies().get("access_token")?.value;

  const sessionId = cloakSSROnlySecret(sessionIdPlain ?? "", "SECRET_KEY");
  return (
    <html lang="en">
      <body className={inter.className}>{children}</body>
    </html>
  );
}
```

**Comment by phryneas** [comment]
Oh wait I just noticed, are they doing `moduleResolution: "bundler"` now? While that's correct, it's also new. I'll have to adjust for that on a TS level. Thanks for the report!

**Comment by phryneas** [comment]
This should be fixed in 0.1.2 now.

**Comment by AlSirang** [comment]
Thank you for the update. The error is fixed now.
